core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -176,7 +176,10 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     packages = Option(packages).orElse(sparkProperties.get("spark.jars.packages")).orNull
     packagesExclusions = Option(packagesExclusions)
       .orElse(sparkProperties.get("spark.jars.excludes")).orNull
-    deployMode = Option(deployMode).orElse(env.get("DEPLOY_MODE")).orNull
+    deployMode = Option(deployMode)
+      .orElse(sparkProperties.get("spark.deployMode"))
```

Comment:
OK, I see. My previous thought is that `spark.submit.deployMode` is only for internal use, so I picked another configuration name. I don't have strong opinion on either one, I will change it.